src/libstore/sqlite.cc

```diff
@@ -258,7 +259,7 @@ void handleSQLiteBusy(const SQLiteBusy & e, time_t & nextWarning)
        is likely to fail again. */
     checkInterrupt();
     /* <= 0.1s */
-    std::this_thread::sleep_for(std::chrono::milliseconds { rand() % 100 });
+    std::this_thread::sleep_for(std::chrono::milliseconds { Random{0, 100}() });
```

Comment:
Can you make `SQLite` require a reference to a `Random` and then for the `LocalStore` use the one you just added, and then add one on `NarInfoDiskCacheImpl`?

Make sure to add some doxygen to the `Random & rng` saying what it is used for :)

Reply:
The function calling that Random, `handleSQLiteBusy()`, isn't a member of the `SQLite` struct. Would it be okay to just keep the functor call here?

Reply:
I think I'll need a bit of help on how to resolve this one. I'm not quite sure where to initialize the `RandomNumberGenerator` when it comes to this specific function. Maybe we could create a `thread_local` global rng for use-cases like this, but is that a suitable solution?